xbmc/threads/Thread.cpp

```diff
@@ -102,6 +102,7 @@ void CThread::Create(bool bAutoDelete, unsigned stacksize)
 
 bool CThread::IsRunning() const
 {
+  CSingleLock lock(m_CriticalSection);
```

Comment:
> to always return the actual state of the thread at any time

And I am trying to say that this does not matter regardless if there is a lock or not. Example, lets assume there is a lock as you suggested:

if (thread->IsRunning())
{
  - we expect the thread as running because IsRunning returned true, but
  - here comes the task switch and thread terminates
  - now we are in a block that is supposed to run if thread is active but it already terminated
}

Even with a lock or atomic the calling thread can't trust the return value if the thread we are querying can terminate.